front/src/applications/stdcm/hooks/useProjectedTrainsForStdcm.ts

```diff
@@ -2,14 +2,62 @@ import { useEffect, useMemo, useState } from 'react';
 
 import { useSelector } from 'react-redux';
 
+import useLazyLoadTrains from 'applications/operationalStudies/hooks/useLazyLoadTrains';
 import type { TrainSpaceTimeData } from 'applications/operationalStudies/types';
 import type { StdcmSuccessResponse } from 'applications/stdcm/types';
 import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 import { useInfraID, useOsrdConfSelectors } from 'common/osrdContext';
 import useLazyProjectTrains from 'modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import { addDurationToDate, subtractDurationFromDate } from 'utils/date';
 
 import formatStdcmTrainIntoSpaceTimeData from '../utils/formatStdcmIntoSpaceTimeData';
 
+/**
+ * Project only the trains which leave 1 hour max before the departure of the stdcm or
+ * arrive 1 hour max after the arrival of the stdcm
+ */
+const keepTrainsRunningDuringStdcm = (
+  stdcmResult: StdcmSuccessResponse | undefined,
```

Comment:
All callers check that `stdcmResult` isn't undefined, so I think we can drop `| undefined` here?